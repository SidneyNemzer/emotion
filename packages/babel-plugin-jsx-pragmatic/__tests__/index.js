import babelTester from 'babel-tester'
import jsxPragmatic from '@emotion/babel-plugin-jsx-pragmatic'
import { transformSync } from '@babel/core'

babelTester('@emotion/babel-plugin-jsx-pragmatic', __dirname, {
  plugins: [
    [
      jsxPragmatic,
      {
        export: 'jsx',
        module: '@emotion/core',
        import: '___EmotionJSX',
        fragment: {
          module: 'react',
          import: 'React'
        }
      }
    ]
  ]
})

test('babel-plugin-jsx-pragmatic should throw error when invalid options', () => {
  expect(() => {
    transformSync('<></>', {
      filename: __filename,
      plugins: [[jsxPragmatic, { export: 'jsx' }]]
    })
  }).toThrow(
    '@emotion/babel-plugin-jsx-pragmatic: You must specify `module` and `import`'
  )
  expect(() => {
    transformSync('<></>', {
      filename: __filename,
      plugins: [
        [
          jsxPragmatic,
          {
            import: '___EmotionJSX',
            module: '@emotion/core',
            fragment: { export: 'default' }
          }
        ]
      ]
    })
  }).toThrow(
    '@emotion/babel-plugin-jsx-pragmatic: You must specify `fragment.module` and `fragment.import`'
  )
})
