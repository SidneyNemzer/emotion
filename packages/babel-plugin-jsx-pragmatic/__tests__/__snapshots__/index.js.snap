// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin-jsx-pragmatic existing-imports 1`] = `
"// inserted import has to go AFTER polyfills
import 'react-app-polyfill/ie11'
import 'react-app-polyfill/stable'

import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

ReactDOM.render(<App />, document.getElementById('root'))


      ↓ ↓ ↓ ↓ ↓ ↓

// inserted import has to go AFTER polyfills
import 'react-app-polyfill/ie11';
import 'react-app-polyfill/stable';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { jsx as ___EmotionJSX } from \\"@emotion/core\\";
ReactDOM.render(<App />, document.getElementById('root'));"
`;

exports[`@emotion/babel-plugin-jsx-pragmatic fragment-only 1`] = `
"import * as React from 'react'

const F = () => <></>


      ↓ ↓ ↓ ↓ ↓ ↓

import * as React from 'react';
import { jsx as ___EmotionJSX } from \\"@emotion/core\\";

const F = () => <></>;"
`;

exports[`@emotion/babel-plugin-jsx-pragmatic fragment-only-no-import 1`] = `
"// When using Emotion, React does not need to be in scope except
// when using fragment shorthand. This plugin should add a react
// import when using the \`fragment\` option, so we can ignore the
// linting error here.
// eslint-disable-next-line react/react-in-jsx-scope
const F = () => <></>


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsx as ___EmotionJSX } from \\"@emotion/core\\";
import { default as React } from \\"react\\";

// When using Emotion, React does not need to be in scope except
// when using fragment shorthand. This plugin should add a react
// import when using the \`fragment\` option, so we can ignore the
// linting error here.
// eslint-disable-next-line react/react-in-jsx-scope
const F = () => <></>;"
`;

exports[`@emotion/babel-plugin-jsx-pragmatic minimal 1`] = `
"import * as React from 'react'

const P = () => <p />


      ↓ ↓ ↓ ↓ ↓ ↓

import * as React from 'react';
import { jsx as ___EmotionJSX } from \\"@emotion/core\\";

const P = () => <p />;"
`;
